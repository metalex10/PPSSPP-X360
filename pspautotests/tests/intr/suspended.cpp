#include <common.h>
#include <pspintrman.h>

extern "C" int main(int argc, char *argv[]) {
	checkpointNext("Not suspended:");
	checkpoint("  0: %08x", sceKernelIsCpuIntrSuspended(0));
	checkpoint("  1: %08x", sceKernelIsCpuIntrSuspended(1));
	checkpoint("  2: %08x", sceKernelIsCpuIntrSuspended(2));
	checkpoint("  0xDEADBEEF: %08x", sceKernelIsCpuIntrSuspended(0xDEADBEEF));

	checkpointNext("Suspended:");
	int flags1 = sceKernelCpuSuspendIntr();
	int flags2 = sceKernelCpuSuspendIntr();
	checkpoint("  0: %08x", sceKernelIsCpuIntrSuspended(0));
	checkpoint("  1: %08x", sceKernelIsCpuIntrSuspended(1));
	checkpoint("  2: %08x", sceKernelIsCpuIntrSuspended(2));
	checkpoint("  0xDEADBEEF: %08x", sceKernelIsCpuIntrSuspended(0xDEADBEEF));
	checkpoint("  flags (inner): %08x", sceKernelIsCpuIntrSuspended(flags2));
	checkpoint("  flags (outer): %08x", sceKernelIsCpuIntrSuspended(flags1));
	sceKernelCpuResumeIntr(flags2);
	sceKernelCpuResumeIntr(flags1);

	return 0;
}