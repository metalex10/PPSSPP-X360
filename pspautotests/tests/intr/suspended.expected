[x] Not suspended:
[x]   0: 00000001
[x]   1: 00000000
[x]   2: 00000000
[x]   0xDEADBEEF: 00000000

[x] Suspended:
[x]   0: 00000001
[x]   1: 00000000
[x]   2: 00000000
[x]   0xDEADBEEF: 00000000
[x]   flags (inner): 00000001
[x]   flags (outer): 00000000
